resources/packs/gcp/gcp.lr

```diff
@@ -1012,32 +1016,134 @@ private gcp.project.cluster.nodepool.config.kubeletConfig @defaults("cpuManagerP
+  // The list of subscriptions in the current project
+  subscriptions() []gcp.project.pubsub.subscription
+  // The list of snapshots in the current project
+  snapshots() []gcp.project.pubsub.snapshot
+}
+
+// GCP PubSub topic
+private gcp.project.pubsub.topic @defaults("name") {
+  // Internal ID
+  id string
+  // Project ID
+  projectId string
+  // The name of the topic
+  name string
+  // The topic configuration
+  config() gcp.project.pubsub.topic.config
+}
+
+// GCP PubSub topic configuration
+private gcp.project.pubsub.topic.config @defaults("kmsKeyName messageStoragePolicy") {
+  // Internal ID
+  id string
+  // The labels associated with this topic
+  labels map[string]string
+  // The name of the Cloud KMS key to be used to protect access to messages published to this topic
+  kmsKeyName string
+  // The message storage policy
+  messageStoragePolicy gcp.project.pubsub.topic.config.messagestoragepolicy
+}
+
+// GCP PubSub topic message storage policy
+private gcp.project.pubsub.topic.config.messagestoragepolicy @defaults("allowedPersistenceRegions"){
+  // Internal ID
+  id string
+  // The list of GCP regions where messages that are published to the topic may be persisted in storage
+  allowedPersistenceRegions []string
+}
+
+// GCP PubSub subscription
+private gcp.project.pubsub.subscription @defaults("name") {
+  // Internal ID
+  id string
+  // Project ID
+  projectId string
+  // The name of the subscription
+  name string
+  // The subscription configuration
+  config() gcp.project.pubsub.subscription.config
+}
+
+// GCP PubSub subscription configuration
+private gcp.project.pubsub.subscription.config @defaults("topic") {
+  // Internal ID
+  id string
+  // The topic to which the subscription points
+  topic gcp.project.pubsub.topic
+  // Configuration for subscriptions that operate in push mode
+  pushConfig gcp.project.pubsub.subscription.config.pushconfig
+  // The default maximum time after a subscriber receives a message before the subscriber should acknowledge the message
+  ackDeadline time
+  // Whether to retain acknowledged messages
+  retainAckedMessages bool
+  // How long to retain messages in backlog from the time of publish
+  retentionDuration time
+  // Specifies the conditions for a subscription's expiration
+  expirationPolicy time
+  // The labels associated with this subscription
+  labels map[string]string
+}
+
+// GCP PubSub Configuration for subscriptions that operate in push mode
+private gcp.project.pubsub.subscription.config.pushconfig {
+  // Internal ID
+  id string
+  // A URL locating the endpoint to which messages should be pushed
+  endpoint string
+  // Endpoint configuration attributes
+  attributes map[string]string
+}
+
+// GCP PubSub snapshot
+private gcp.project.pubsub.snapshot @defaults("name") {
+  // Internal ID
+  id string
+  // Project ID
+  projectId string
+  // The name of the subscription
+  name string
+  // The topic for which the snapshot is
```

Comment:
Hey @imilchev is this the original topic that this is a snapshot of?

Reply:
I believe so. All docs I took from the documentation of the Google API. I don't know all of those concepts myself either... 😞

Reply:
Aw, please don't be 😞 ! What if we change this to "The original topic of which this is a snapshot" ? I think that would be more helpful.